spring-integration-core/src/main/java/org/springframework/integration/annotation/MessagingGateway.java

```diff
@@ -68,7 +68,7 @@
 	 * For example if this gateway is hooked up to a {@code QueueChannel}. 
 	 * @return the suggested timeout in milliseconds, if any
 	 */
-	long defaultRequestTimeout() default Long.MIN_VALUE;
+	String defaultRequestTimeout() default "9223372036854775807";
```

Comment:
OK. Just let me know, if you don't have anything specific to this point.
And I'll fix on merge or leave as is.
Just convince me that this change is correct or not :smile: